src/Gui/QSint/actionpanel/actionpanel.cpp

```diff
@@ -11,110 +11,84 @@
 
 #include <QVariant>
 
-
 namespace QSint
 {
 
-
 ActionPanel::ActionPanel(QWidget *parent) :
     BaseClass(parent), mySpacer(nullptr)
 {
     setProperty("class", "panel");
-
     setScheme(ActionPanelScheme::defaultScheme());
-
     setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Preferred);
 
-    QVBoxLayout *vbl = new QVBoxLayout();
+    auto *vbl = new QVBoxLayout();
     vbl->setContentsMargins(4, 8, 4, 8);
     vbl->setSpacing(8);
     setLayout(vbl);
 }
 
 void ActionPanel::setScheme(ActionPanelScheme *scheme)
 {
-  if (scheme) {
+    if (!scheme) return;
+
     myScheme = scheme;
     setStyleSheet(myScheme->actionStyle);
 
-    // set scheme for children
-    QObjectList list(children());
-    Q_FOREACH(QObject *obj, list) {
-      if (dynamic_cast<ActionGroup*>(obj)) {
-        ((ActionGroup*)obj)->setScheme(scheme);
-        continue;
-      }
-    }
-
     update();
-  }
 }
 
-//void ActionPanel::paintEvent ( QPaintEvent * event )
-//{
-//  //QPainter p(this);
-
-//  //p.setOpacity(0.5);
-//  //p.fillRect(rect(), myScheme->panelBackground);
-//}
-
 void ActionPanel::addWidget(QWidget *w)
 {
-  if (w)
-    layout()->addWidget(w);
+    if (w) layout()->addWidget(w);
 }
 
 void ActionPanel::removeWidget(QWidget *w)
 {
-  if (w)
-    layout()->removeWidget(w);
+    if (w) layout()->removeWidget(w);
 }
 
-void ActionPanel::addStretch(int s)
+void ActionPanel::addStretch(int /*s*/)
 {
-  Q_UNUSED(s); 
-  //((QVBoxLayout*)layout())->addStretch(s);
-  if (!mySpacer) {
-    mySpacer = new QSpacerItem(0,0,QSizePolicy::Minimum, QSizePolicy::Expanding);
-    layout()->addItem(mySpacer);
-  }
+    if (!mySpacer) {
+        mySpacer = new QSpacerItem(0, 0, QSizePolicy::Minimum, QSizePolicy::Expanding);
+        layout()->addItem(mySpacer);
+    }
 }
 
 void ActionPanel::removeStretch()
 {
-  if (mySpacer) {
-    layout()->removeItem(mySpacer);
-    delete mySpacer;
-    mySpacer = nullptr;
-  }
+    if (mySpacer) {
+        layout()->removeItem(mySpacer);
+        delete mySpacer;
+        mySpacer = nullptr;
+    }
 }
 
-ActionGroup * ActionPanel::createGroup()
+ActionGroup *ActionPanel::createGroup()
 {
-    ActionGroup * group = new ActionGroup(this);
+    auto *group = new ActionGroup(this);
```

Comment:
Would prefer if we forward on didn't change to `auto` without it improving readability - just to avoid git noise

Reply:
> That one removes clear repetition of the tyle

True, but we only want to avoid repetition if it adds value, e.g. improves readability and maintainability - not for the repetitions own sake. Whether it's `ActionGroup` is repeated or it's `auto` instead doesn't change readability or maintainability.

Anyways, let's not worry about it for now, just keep the `auto` in the PR.